File: summary/ruby/6_prime_number.rb
def is_prime(n)
  return false if n <= 1

  limit = Math.sqrt(n)
  (2..limit).each do |i|
    return false if n % i == 0
  end

  true
end

arr = gets.chomp.split.map(&:to_i)
arr.each do |v|
  if is_prime(v)
    puts "#{v} は素数です。"
  else
    puts "#{v} は素数ではありません。"
  end
end

File: summary/rust/6_prime_number.rs
fn is_prime(n: u64) -> bool {
  if n <= 1 {
    return false;
  }

  for i in 2..n {
    if n % i == 0 {
      return false;
    }
  }

  true
}

fn main() {
    println!("{}", is_prime(7)); // true
    println!("{}", is_prime(10)); // false
    println!("{}", is_prime(23)); // true
}
